ramalama/cli.py

```diff
@@ -751,6 +764,12 @@ def serve_parser(subparsers):
     parser.add_argument(
         "-p", "--port", default=config.get('port', "8080"), help="port for AI Model server to listen on"
     )
+    parser.add_argument(
+        "--network-mode",
+        type=str,
+        default="bridge",
```

Comment:
@rhatdan @rhjostone I am wondering if this should be private instead of bridge. The private creates the bridge.

Reply:
Let's make this `private`, since I believe `bridge` will work differently in `rootless` containers. `private` allows the container engine to make the choice.  If the network is set to private, on Docker then we would need to to force bridge, since `private` is not understood on Docker.

@Luap99 Correct me if I am wrong?

Reply:
Looks like bridge works for rootless podman ok, so will merge.

Reply:
`private` would be my preferred choice since it works for both rootful and rooltess podman:

```
private: Create a new namespace for the container. This uses the bridge mode for rootful containers and slirp4netns for rootless ones.
```

Docker however doesn't appear to have a `private` mode:

https://docs.docker.com/engine/network/#drivers

Since ramalama supports both podman and docker, I wanted to use whatever option would work between all engines. Users can at least still explicitly set `private` for more confined networking.

Reply:
You should be able to request `default` instead, I think that works on docker too. I don't think bridge is a great default for rootless podman. Using the direct pasta driver has more advantages, keeps source ip and is a less complex network setup. The only advantage of bridge is that other containers can connect directly to it without going through host ports.

Reply:
That's a good idea. I've changed the default value to `default`.